io.openems.edge.controller.ess.limitusablecapacity/src/io/openems/edge/controller/ess/limitusablecapacity/LimitUsableCapacityControllerImpl.java

```diff
@@ -0,0 +1,181 @@
+package io.openems.edge.controller.ess.limitusablecapacity;
+
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.exceptions.OpenemsException;
+import io.openems.edge.common.channel.value.Value;
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.ComponentManager;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.controller.api.Controller;
+import io.openems.edge.ess.api.ManagedSymmetricEss;
+import io.openems.edge.ess.power.api.Constraint;
+import io.openems.edge.ess.power.api.Phase;
+import io.openems.edge.ess.power.api.Pwr;
+import io.openems.edge.ess.power.api.Relationship;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(//
+		name = "Controller.Ess.LimitUsableCapacity", //
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE //
+)
+public class LimitUsableCapacityControllerImpl extends AbstractOpenemsComponent
+		implements LimitUsableCapacityController, Controller, OpenemsComponent {
+
+	@Reference
+	protected ComponentManager componentManager;
+
+	// Force charge power
+	private int forceChargePower = 2000;
+
+	private Config config;
+	private State state = State.UNDEFINED;
+
+	public LimitUsableCapacityControllerImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				Controller.ChannelId.values(), //
+				LimitUsableCapacityController.ChannelId.values() //
+		);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) throws OpenemsException {
+		super.activate(context, config.id(), config.alias(), config.enabled());
+
+		this.config = config;
+
+		/*
+		 * Checking the Soc values in the configuration
+		 * 
+		 * forceChargeSoc < stopDischargeSoc < allowDischargeSoc < allowChargeSoc <
+		 * stopChargeSoc
+		 * 
+		 */
+		if (this.config.forceChargeSoc() > this.config.stopDischargeSoc()
+				&& this.config.stopDischargeSoc() > this.config.allowDischargeSoc()
+				&& this.config.allowDischargeSoc() > this.config.allowChargeSoc()
+				&& this.config.allowChargeSoc() > this.config.stopChargeSoc()) {
+			throw new OpenemsException("Please re-check the configuration, invalid values present in the Soc columns");
+		}
+	}
+
+	@Override
+	public void run() throws OpenemsNamedException {
+		ManagedSymmetricEss ess = this.componentManager.getComponent(this.config.ess_id());
```

Comment:
It would be preferrable to use `@Reference` for getting the `ManagedSymmetricEss`.

I discussed also with @JohannesHundschell, that it would be good to allow having this Controller applied on multiple ESS at the same time. This avoids having to configure it for each and every ESS in a big system separately. Can you implement that or do you need support?

Reply:
Here is an example for `@Reference` with multiple Component-IDs:

https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.meter.virtual/src/io/openems/edge/meter/virtual/symmetric/subtract/VirtualSubtractMeter.java#L65-L69

Reply:
Changed with ManagedSymmetricEss reference.